doc/user/content/transform-data/patterns/rules-engine.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "Rules execution engine"
+description: "Encode rules as data and apply them using lateral joins."
+aliases:
+  - /guides/rules-engine/
+  - /sql/patterns/rules-engine/
+menu:
+  main:
+    parent: 'sql-patterns'
+---
+
+SQL queries are often used to represent "rules" and evaluate which data in a dataset adhere to those rules.
+In Materialize, the results of these queries can be kept automatically up-to-date, enabling use cases like real-time, rules-based alerting.
+However, some organizations have a lot of rules.
+This can lead users to ask questions like "Can Materialize support 100,000 materialized views"?
```

Comment:
```suggestion
Some organizations may need more rules to support thousands of materialized views.
```

Reply:
We're trying to avoid having to write and maintain thousands of materialized views! Gonna reword this a bit:
> Rule execution use cases can have many thousands of rules, so it's not practical to have a separate view for each rule.

Reply:
Ah I see now! Thanks for the clarification